File: practica Gral/polimorfismo/Matriz.py
#clase base
    def __init__(self, valores=None):
        self.matriz = [[0.0 for _ in range(10)] for _ in range(10)]

        if valores is None:
            for i in range(10):
                for j in range(10):
                    if i == j:
                        self.matriz[i][j] = 1.0
                    else:
                        self.matriz[i][j] = 0.0
        else:
            for i in range(10):
                for j in range(10):
                    self.matriz[i][j] = valores[i][j]

    def mostrar(self):
        for i in range(10):
            for j in range(10):
                print(f"{self.matriz[i][j]:.1f}", end="\t")
            print()
        print()

    def sumar(self, m):
        resultado = [[0.0 for _ in range(10)] for _ in range(10)]
        for i in range(10):
            for j in range(10):
                resultado[i][j] = self.matriz[i][j] + m.matriz[i][j]
        return Matriz(resultado)

    def restar(self, m):
        resultado = [[0.0 for _ in range(10)] for _ in range(10)]
        for i in range(10):
            for j in range(10):
                resultado[i][j] = self.matriz[i][j] - m.matriz[i][j]
        return Matriz(resultado)

    def igual(self, m):
        for i in range(10):
            for j in range(10):
                if self.matriz[i][j] != m.matriz[i][j]:
                    return False
        return True


# Main
if __name__ == "__main__":
    m1 = Matriz()
    print("Matriz identidad:")
    m1.mostrar()

    # Crear matriz personalizada
    valores = [[i + j for j in range(10)] for i in range(10)]
    m2 = Matriz(valores)

    print("Matriz personalizada:")
    m2.mostrar()

    suma = m1.sumar(m2)
    print("Suma de matrices:")
    suma.mostrar()

    resta = m2.restar(m1)
    print("Resta de matrices:")
    resta.mostrar()

    print("m1 igual a m2?", m1.igual(m2))
    print("m1 igual a m1?", m1.igual(m1))

File: practica Gral/polimorfismo/PoliEj3/Matriz.java
package PoliEj3;
public class Matriz {
    private float[][] matriz;

    public Matriz() {
        matriz = new float[10][10];
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (i == j) {
                    matriz[i][j] = 1; 
                } else {
                    matriz[i][j] = 0; 
                }
            }
        }
    }

    public Matriz(float[][] valores) {
        matriz = new float[10][10];
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                matriz[i][j] = valores[i][j];
            }
        }
    }

    public void mostrar() {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                System.out.print(matriz[i][j] + "\t");
            }
            System.out.println();
        }
        System.out.println();
    }

    public Matriz sumar(Matriz m) {
        float[][] resultado = new float[10][10];
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                resultado[i][j] = this.matriz[i][j] + m.matriz[i][j];
            }
        }
        return new Matriz(resultado);
    }

    public Matriz restar(Matriz m) {
        float[][] resultado = new float[10][10];
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                resultado[i][j] = this.matriz[i][j] - m.matriz[i][j];
            }
        }
        return new Matriz(resultado);
    }
    public boolean igual(Matriz m) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (this.matriz[i][j] != m.matriz[i][j]) {
                    return false;
                }
            }
        }
        return true;
    }
}

